crypto/src/lib.rs

```diff
@@ -547,37 +456,152 @@ impl PrivateKey {
     }
 }
 
-pub mod ffi {
+pub mod error {
+    //! Module containing errors
+    use super::*;
+
+    /// Error indicating algorithm could not be found
+    #[derive(Debug, Display, Clone, Copy)]
+    #[display(fmt = "Algorithm not supported")]
+    pub struct NoSuchAlgorithm;
+
+    #[cfg(feature = "std")]
+    impl std::error::Error for NoSuchAlgorithm {}
+
+    /// Error which occurs when converting to/from `Multihash`
+    #[derive(Debug, Clone, Display)]
+    pub struct MultihashConvertError {
+        reason: String,
+    }
+
+    impl MultihashConvertError {
+        #[cfg(not(feature = "ffi_import"))]
+        pub(crate) const fn new(reason: String) -> Self {
+            Self { reason }
+        }
+    }
+
+    #[cfg(feature = "std")]
+    impl std::error::Error for MultihashConvertError {}
+
+    impl From<NoSuchAlgorithm> for MultihashConvertError {
+        fn from(_: NoSuchAlgorithm) -> Self {
+            Self {
+                reason: String::from("Digest function not supported"),
+            }
+        }
+    }
+
+    /// Error which occurs when parsing [`PublicKey`]
+    #[derive(Debug, Display, Clone)]
+    pub enum KeyParseError {
+        /// Decoding hex failed
+        #[display(fmt = "Failed to decode. {_0}: {_1}")]
+        Decode(String, hex::FromHexError),
```

Comment:
```suggestion
        Decode{
          candidate: String, 
          error: hex::FromHexError
        },
```

This would make it slightly more readable IMO.

Reply:
this is no longer actual. I'm closing